server/src/main/java/org/elasticsearch/common/network/ThreadWatchdog.java

```diff
@@ -182,12 +189,14 @@ private final class Checker extends AbstractRunnable {
         private final TimeValue interval;
         private final TimeValue quietTime;
         private final Lifecycle lifecycle;
+        private final Logger logger;
 
-        Checker(ThreadPool threadPool, TimeValue interval, TimeValue quietTime, Lifecycle lifecycle) {
+        Checker(ThreadPool threadPool, TimeValue interval, TimeValue quietTime, Lifecycle lifecycle, Logger logger) {
             this.threadPool = threadPool;
             this.interval = interval;
             this.quietTime = quietTime.compareTo(interval) <= 0 ? interval : quietTime;
             this.lifecycle = lifecycle;
+            this.logger = logger;
```

Comment:
The hot-threads dumper still links the doc for `ReferenceDocs.NETWORK_THREADING_MODEL` which is not applicable in the new case. Do we care?